homeassistant/components/media_player/androidtv.py

```diff
@@ -0,0 +1,488 @@
+"""
+Provide functionality to interact with AndroidTv devices on the network.
+
+Example config using an external ADB server:
+media_player:
+  - platform: androidtv
+    host: 192.168.1.37
+    name: MIBOX3
+    adb_server_ip: 127.0.0.1
+    adb_server_port: 5037
+    apps:
+      "amazon": "Amazon Premium Video"
+
+Example config using purely Python:
+media_player:
+  - platform: androidtv
+    host: 192.168.1.37
+    name: MIBOX3
+    adbkey: /config/adbkey
+    apps:
+      "amazon": "Amazon Premium Video"
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.androidtv/
+"""
+
+import logging
+import functools
+import os
+import threading
+import voluptuous as vol
+
+from homeassistant.components.media_player import (
+    DOMAIN, MediaPlayerDevice, PLATFORM_SCHEMA, SUPPORT_NEXT_TRACK,
+    SUPPORT_PAUSE, SUPPORT_PLAY,
+    SUPPORT_PREVIOUS_TRACK, SUPPORT_STOP, SUPPORT_TURN_OFF,
+    SUPPORT_TURN_ON, SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_STEP)
+
+from homeassistant.const import (
+    ATTR_ENTITY_ID, CONF_HOST, CONF_NAME, CONF_PORT,
+    STATE_IDLE, STATE_PAUSED, STATE_PLAYING, STATE_OFF)
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['androidtv==0.0.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_APPS = 'apps'
+CONF_ADBKEY = 'adbkey'
+CONF_ADB_SERVER_IP = 'adb_server_ip'
+CONF_ADB_SERVER_PORT = 'adb_server_port'
+
+DEFAULT_APPS = {}
+DEFAULT_ADBKEY = os.path.join(os.path.expanduser('~'), '.android', 'adbkey')
+DEFAULT_NAME = 'Android'
+DEFAULT_PORT = '5555'
+DEFAULT_ADB_SERVER_PORT = 5037
+
+
+def has_adb_files(value):
+    """Check that ADB key files exist."""
+    priv_key = value
+    pub_key = '{}.pub'.format(value)
+    cv.isfile(pub_key)
+    return cv.isfile(priv_key)
+
+
+ACTIONS = {
+    "back": "4",
```

Comment:
I would prefer to let it here because as it used for a service, there is a check to see if the requested action exists in the component. Also contributors can more easily add their actions in the component rather than in the backend lib.